File: 53-maximum-subarray/maximum-subarray.c
int maxSubArray(int* n, int s) {
    int max=-10000000;
    int sum=0;
        for(int i=0;i<s;i++){
            sum=sum+n[i];
            if(sum>max){
                max=sum;
            }
            if(sum<0)
            sum=0;
        }
        return max;
}

File: 53-maximum-subarray/maximum-subarray.java
class Solution {
    public int maxSubArray(int[] arr) {
        int s=0;
        int max=Integer.MIN_VALUE;
        for(int i=0;i<arr.length;i++){
            s+=arr[i];
            if(max<s)max=s;
            if(s<0)s=0;
        }
        return max;
    }
}